lua/orgmode/org/mappings.lua

```diff
@@ -246,8 +246,40 @@ function OrgMappings:handle_return(suffix)
       return vim.cmd([[startinsert!]])
     end
     if number_in_list then
-      vim.fn.append(vim.fn.line('.'), string.format('%s%d%s ', indent, tonumber(number_in_list) + 1, closer))
-      vim.fn.cursor(vim.fn.line('.') + 1, 0)
+      local next_sibling = item.node:parent()
+      local text_edits = {}
+      local counter = 1
+      local new_start
+      while next_sibling do
+        if next_sibling:type() == 'listitem' then
```

Comment:
Can this ever not be `listitem`? Does it make sense to break out if this is false?

Reply:
`next_sibling` returns all nodes, named and unamed. So it can be the newline at the end of each list item.
I'll change it to `ts_utils.get_next_node`, that only returns named nodes, that is easier to understand.